### PR TITLE
Add rx.Completable support for ContentResolver

### PR DESCRIPTION
**Pull request by geralt-encore**
https://github.com/pushtorefresh/storio/issues/628

### CODE REVIEW

**Comment by geralt-encore** [comment]
Something wrong with Travis again I guess

**Comment by nikitin-da** [comment]
Yeah, smth with coverage report was wrong

**Comment by nikitin-da** [comment]
Awesome! :+1:

File: storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java [hunk 1]
```diff
@@ -126,6 +128,25 @@ public DeleteResult executeAsBlocking() {
...
+     * <dt><b>Scheduler:</b></dt>
+     * <dd>Operates on {@link Schedulers#io()}.</dd>
+     * </dl>
+     *
+     * @return non-null {@link Completable} which will perform Delete Operation.
+     */
+    @NonNull
+    @CheckResult
+    @Override
+    public Completable asRxComletable() {
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: Wait, we merged wrong method name?? (reviewing from phone)

Comment (geralt-encore) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: Nope, no typo here (it was "compleatable" before fixing)

Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: I see typo: letter `p` is missing

Comment (geralt-encore) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: facepalm

Comment (geralt-encore) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: Sorry, my bad =(

Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: :D np

File: storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java [hunk 2]
```diff
@@ -126,6 +128,25 @@ public DeleteResult executeAsBlocking() {
...
+     * <dd>Operates on {@link Schedulers#io()}.</dd>
+     * </dl>
+     *
+     * @return non-null {@link Completable} which will perform Delete Operation.
+     */
+    @NonNull
+    @CheckResult
+    @Override
+    public Completable asRxCompletable() {
+        throwExceptionIfRxJavaIsNotAvailable("asRxCompetable()");
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: `p` is missing in the exception message :)

Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: shit, I meant: `l`

Comment (geralt-encore) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: shit looks like I have a serious problem with pronunciation of this word =)

Comment (artem-zinnatullin) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteByQuery.java]: :D (pls check same thing in StorIOSQLite)

File: storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteObjectTest.java [hunk 3]
```diff
@@ -203,5 +231,36 @@ public void shouldThrowExceptionIfNoTypeMappingWasFoundWithoutAffectingContentPr
...
+                    .object(TestItem.newInstance())
+                    .prepare()
+                    .asRxCompletable()
+                    .subscribe(testSubscriber);
+
+            testSubscriber.awaitTerminalEvent();
+            testSubscriber.assertNoValues();
+            assertThat(testSubscriber.getOnErrorEvents().get(0))
+                    .isInstanceOf(StorIOException.class)
+                    .hasCauseInstanceOf(IllegalStateException.class);
```
Comment (artem-zinnatullin) [inline storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/operations/delete/PreparedDeleteObjectTest.java]: nit: Will be great to check exception message here and some similar tests below

**Comment by artem-zinnatullin** [comment]
@geralt-encore please squash PR and I'll merge it!

**Comment by geralt-encore** [comment]
Squashed

**Comment by artem-zinnatullin** [comment]
:+1: thanks a lot!

// sorry for delay with merging, I was in a trip

**Pull request merged**